src/System.Collections.Specialized/src/System/Collections/Specialized/ListDictionary.cs

```diff
@@ -493,6 +492,7 @@ public void Reset()
         }
 
         [Serializable]
+        [System.Runtime.CompilerServices.TypeForwardedFrom("System, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089")]
```

Comment:
They are at least needed on types which live in mscorlib on netfx and are not in System.Private.Corelib on core for the optimization of BinaryFormatter (strip assembly full names).